efficientdet/keras/efficientdet_arch_keras_test.py

```diff
@@ -54,31 +54,28 @@ def test_model_output(self):
       feats = efficientdet_arch_keras.build_backbone(feats, config)
       feats = efficientdet_arch_keras.build_feature_network(feats, config)
       feats = efficientdet_arch_keras.build_class_and_box_outputs(feats, config)
+      # TODO(tanmingxing): Fix the failure for keras Model.
```

Comment:
I try to use a more scientific way to test model unify:
```
  def test_model_output(self):
    inputs_shape = (1, 512, 512, 3)
    config = hparams_config.get_efficientdet_config('efficientdet-d0')
    ckpt_path = '/tmp/test_ckpt'

    with tf.Session(graph=tf.Graph()) as sess:
      feats = tf.ones(inputs_shape)
      feats = legacy_arch.efficientdet(feats, config=config)
      sess.run(tf.global_variables_initializer())
      legacy_class_out, legacy_box_out = sess.run(feats)
      all_saver = tf.train.Saver()
      all_saver.save(sess, ckpt_path)

    ckpt = tf.train.load_checkpoint(ckpt_path)
    with tf.Session(graph=tf.Graph()) as sess:
      feats = tf.ones(inputs_shape)
      feats = efficientdet_arch_keras.build_backbone(feats, config)
      feats = efficientdet_arch_keras.build_feature_network(feats, config)
      feats = efficientdet_arch_keras.build_class_and_box_outputs(feats, config)
      variables = tf.global_variables()
      assign_ops = [var.assign(ckpt.get_tensor(var.name.split(':')[0])) for var in variables]

      sess.run(assign_ops)
      keras_class_out, keras_box_out = sess.run(feats)

    for i in range(3, 8):
      self.assertAllClose(keras_class_out[i - 3], legacy_class_out[i], rtol=1e-4, atol=1e-4)
      self.assertAllClose(keras_box_out[i - 3], legacy_box_out[i], rtol=1e-4, atol=1e-4)
    feats = tf.ones(inputs_shape)
    model = efficientdet_arch_keras.EfficientDetModel(config=config)
    model.build(inputs_shape)
    [var.assign(ckpt.get_tensor(var.name.split(':')[0])) for var in model.variables]
    eager_class_out, eager_box_out = model(feats)
    for i in range(3, 8):
      # TODO(tanmingxing): fix the failing case.
      self.assertAllClose(eager_class_out[i - 3], legacy_class_out[i], rtol=1e-4, atol=1e-4)
      self.assertAllClose(eager_box_out[i - 3], legacy_box_out[i], rtol=1e-4, atol=1e-4)
```
I found there are some tolerances in ```build_feature_network``` when I load the efficientdet-d0 checkpoint.
Could you check it?

Reply:
yes, if the weights are  loaded from a checkpoint, then the outputs are close. But if the weights are initialized from random values (with the same random seed), then the outputs are different. I suspect that eager mode initializes variables in different order compared to graph mode.